src/device-state/db-format.ts

```diff
@@ -92,67 +88,77 @@ export async function setApps(
 /**
  * Create target state from database state
  */
-export async function getTargetJson(): Promise<TargetApps> {
+export async function getTargetWithRejections(): Promise<{
+	apps: TargetApps;
+	rejections: AppRelease[];
+}> {
 	const dbApps = await getDBEntry();
 
-	return dbApps
-		.map(
-			({
-				source,
-				uuid,
-				releaseId,
-				commit: releaseUuid,
-				...app
-			}): [string, TargetApp] => {
-				const services = (JSON.parse(app.services) as DatabaseService[])
-					.map(
-						({
-							serviceName,
-							serviceId,
-							imageId,
-							...service
-						}): [string, TargetService] => [
-							serviceName,
-							{
-								id: serviceId,
-								image_id: imageId,
-								..._.omit(service, ['appId', 'appUuid', 'commit', 'releaseId']),
-							} as TargetService,
-						],
-					)
-					// Map by serviceName
-					.reduce(
-						(svcs, [serviceName, s]) => ({
-							...svcs,
-							[serviceName]: s,
-						}),
-						{},
-					);
-
-				const releases = releaseUuid
-					? {
-							[releaseUuid]: {
-								id: releaseId,
-								services,
-								networks: JSON.parse(app.networks),
-								volumes: JSON.parse(app.volumes),
-							} as TargetRelease,
-						}
-					: {};
-
-				return [
-					uuid,
+	const apps: TargetApps = {};
+	const rejections: AppRelease[] = [];
+
+	for (const {
+		source,
+		rejected,
+		uuid,
+		releaseId,
+		commit: releaseUuid,
+		...app
+	} of dbApps) {
+		const services = Object.fromEntries(
+			(JSON.parse(app.services) as DatabaseService[]).map(
+				({
+					serviceName,
+					serviceId,
+					imageId,
+					// Ignore these fields
+					appId: _appId,
+					appUuid: _appUuid,
+					commit: _commit,
+					releaseId: _releaseId,
```

Comment:
In typescript an `_` prefix has a very specific meaning which is to ignore an unused variable, that is not the case for these so I'd like to see some other prefix and the standard I tend to use as a placeholder prefix is `$`:
```suggestion
					appId: $appId,
					appUuid: $appUuid,
					commit: $commit,
					releaseId: $releaseId,
```

Reply:
But these variables are unused, that's why I used the prefix

Reply:
Right, I didn't spot that this was a destructuring, my bad